runtime/browser/web_application.cc

```diff
@@ -485,13 +484,74 @@ void WebApplication::Terminate() {
   }
 }
 
+void WebApplication::CreateWindow(common::AppControl* appcontrol) {
+  auto window_category = GetWindowCategory(appcontrol);
+  window_.reset(CreateNativeWindow(window_category));
```

Comment:
This method is called only and only when window creation is needed. If the same category has been deduced in AppControl(), this method should not be called.